src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -315,7 +303,11 @@ private void expectField(JsonParser jsonParser, String expectedFieldName) throws
     private void expectToken(JsonParser jsonParser, JsonToken expectedToken) throws IOException {
         final JsonToken token = jsonParser.nextToken();
         if (token == null) {
-            throw new EOFException(Thread.currentThread().isInterrupted() ? "Thread was interrupted" : "Stream was closed");
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
Doesn't it need to be

```
if (Thread.currentThread().isInterrupted()) {
    throw new InterruptedIOException("Thread was interrupted");
} else {
    if (token == null) {
        throw new EOFException("Stream was closed");
}
```
?
Otherwise the stream won't stop until you also close the connection.